tracer/test/Datadog.Trace.Tools.dd_dotnet.ArtifactTests/CreatedumpTests.cs

```diff
@@ -0,0 +1,343 @@
+// <copyright file="CreatedumpTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if !NETFRAMEWORK
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.Tools.dd_dotnet.ArtifactTests;
+
+public class CreatedumpTests : ConsoleTestHelper
+{
+    private const string CreatedumpExpectedOutput = "Writing minidump";
+    private const string CrashReportExpectedOutput = "The crash might have been caused by automatic instrumentation";
+
+    public CreatedumpTests(ITestOutputHelper output)
+        : base(output)
+    {
+        // Those environment variables can be set by Nuke, and will impact the outcome of the tests
+        EnvironmentHelper.CustomEnvironmentVariables["COMPlus_DbgMiniDumpType"] = string.Empty;
+        EnvironmentHelper.CustomEnvironmentVariables["COMPlus_DbgEnableMiniDump"] = string.Empty;
+        EnvironmentHelper.CustomEnvironmentVariables["DD_INSTRUMENTATION_TELEMETRY_ENABLED"] = string.Empty;
+    }
+
+    private static (string Key, string Value) LdPreloadConfig
+    {
+        get
+        {
+            var path = Utils.GetApiWrapperPath();
+
+            if (!File.Exists(path))
+            {
+                throw new FileNotFoundException($"LD wrapper not found at path {path}");
+            }
+
+            return ("LD_PRELOAD", path);
+        }
+    }
+
+    private static (string Key, string Value)[] CreatedumpConfig => [("COMPlus_DbgEnableMiniDump", "1"), ("COMPlus_DbgMiniDumpName", "/dev/null")];
+
+    [SkippableTheory]
+    [InlineData("1", true)]
+    [InlineData("0", false)]
+    public async Task Passthrough(string passthrough, bool shouldCallCreatedump)
+    {
+        // This tests the case when a dotnet exe calls another one
+        // The expected behavior is that in the parent process, we check if COMPlus_DbgEnableMiniDump
+        // was already set. If it was, then we need to forward the call to createdump in case of crash.
+        // If it wasn't set, then we set it so that dd-dotnet will be invoked in case of crash.
+        // If a child dotnet process is spawned, we may then mistakenly think that COMPlus_DbgEnableMiniDump
+        // was set from the environment, even though it was set by us. To prevent that, we set the
+        // DD_TRACE_CRASH_HANDLER_PASSTHROUGH environment variable, which codifies the result of the
+        // "was COMPlus_DbgEnableMiniDump set?" check.
+
+        SkipOn.Platform(SkipOn.PlatformValue.MacOs);
```

Comment:
None of this works on Windows either, right? If it does, we should add the `RunOnWindows` trait so that we test it

Reply:
Yeah it's linux only.